prometheus_client/openmetrics/exposition.py

```diff
@@ -59,16 +69,40 @@ def generate_latest(registry):
                 timestamp = ''
                 if s.timestamp is not None:
                     timestamp = f' {s.timestamp}'
-                output.append('{}{} {}{}{}\n'.format(
-                    s.name,
-                    labelstr,
-                    floatToGoString(s.value),
-                    timestamp,
-                    exemplarstr,
-                ))
+                if _is_valid_legacy_metric_name(s.name):
+                    output.append('{}{} {}{}{}\n'.format(
+                        s.name,
+                        labelstr,
+                        floatToGoString(s.value),
+                        timestamp,
+                        exemplarstr,
+                    ))
+                else:
+                    output.append('{} {}{}{}\n'.format(
+                        labelstr,
+                        floatToGoString(s.value),
+                        timestamp,
+                        exemplarstr,
+                    ))
         except Exception as exception:
             exception.args = (exception.args or ('',)) + (metric,)
             raise
 
     output.append('# EOF\n')
     return ''.join(output).encode('utf-8')
+
+
+def escape_metric_name(s: str) -> str:
+    if _is_valid_legacy_metric_name(s):
+        return s
+    return '"{}"'.format(escape(s))
+
+
+def escape_label_name(s: str) -> str:
+    if _is_valid_legacy_labelname(s):
+        return s
+    return '"{}"'.format(escape(s))
+
+
+def escape(s: str) -> str:
```

Comment:
This one as private could be nice s well. I like the other escapes for metric and label names being public though as part of this work.